DSharpPlus/Net/RateLimitBucket.cs

```diff
@@ -41,7 +41,7 @@ public string BucketId
         /// <summary>
         /// Gets the number of uses left before pre-emptive rate limit is triggered.
         /// </summary>
-        public int Remaining { get; internal set; }
+        public int Remaining => _remaining;
```

Comment:
If you're introducing a backing field to enable atomic writes, you should also make the property to perform an atomic read, preferably via `Volatile` or `Interlocked` classes.

Reply:
All 32-bit reads are atomic (unless you are on some strange architecture or a 16-bit system but I am pretty sure .net does not run on such a platform). Making it a volatile read would be possible but the only place where it really matters is the check if there are requests available but that is implemented via the atomic decrement. The only place that uses that property now is the ToString method and there it probably won't matter if that is the most recent value as it will only be visible in the debugger